packages/packages.go

```diff
@@ -28,12 +32,22 @@ type Packages interface {
 // FIXME: Research the various Gopkg name formats and convert them correctly
 func convertGopkgNameToPurl(name string) string {
 	switch {
-	case strings.Contains(name, "github.com"):
-		name = strings.Replace(name, "github.com", "github", 1)
-	case strings.Contains(name, "gopkg.in"):
-		name = strings.Replace(name, "gopkg.in", "github", 1)
-	case strings.Contains(name, "golang.org"):
-		name = strings.Replace(name, "golang.org", "golang", 1)
+	case githubPattern.MatchString(name):
+		// Currently OSS Index's github format support is based on repository
+		// owner/name, so restrict the PURL name path.
+		//
+		// Once golang format support is improved we can switch to that and get
+		// more precise and refined results on repository sub-paths
+		name = githubPattern.ReplaceAllString(name, "github/$1/$2")
+
+	case gopkg2Pattern.MatchString(name):
+		name = gopkg2Pattern.ReplaceAllString(name, "github/$1/$2")
```

Comment:
QUESTION: Not knowing enough here, but this uses the gopk2Pattern, which is: `("^gopkg.in/([^/]+)/([^.]+).*")`, should it be stuffing `github` in the replace?

Reply:
gopkg.in is a service that maps shortened URLs with github repositories following a certain pattern, so we are just following the same pattern to find the real URLs.